Examples/Io/Csv/src/CsvMeasurementReader.cpp

```diff
@@ -112,14 +129,62 @@ std::vector<ActsExamples::MeasurementData> readMeasurementsByGeometryId(
   return measurements;
 }
 
-std::vector<ActsExamples::CellData> readCellsByHitId(
-    const std::string& inputDir, size_t event) {
-  // timestamp is an optional element
-  auto cells = readEverything<ActsExamples::CellData>(inputDir, "cells.csv",
-                                                      {"timestamp"}, event);
-  // sort for fast hit id look up
-  std::sort(cells.begin(), cells.end(), CompareHitId{});
-  return cells;
+// Build clusters from the cell data vector. Assumes the cell data to be sorted
+// by measurment id!
+ActsExamples::ClusterContainer makeClusters(
+    const std::vector<ActsExamples::CellData>& cellData,
+    std::size_t nMeasurments) {
+  using namespace ActsExamples;
+  ClusterContainer clusters;
+
+  auto cellDataIt = cellData.begin();
+  for (auto index = 0ul; index < nMeasurments; ++index) {
+    // Search for the range containing the index
+    cellDataIt = std::find_if(cellDataIt, cellData.end(), [&](const auto& cd) {
+      return cd.measurement_id == index;
+    });
+
+    auto nextCellDataIt = std::find_if(
+        cellDataIt, cellData.end(),
+        [&](const auto& cd) { return cd.measurement_id != index; });
```

Comment:
would this be a case for `upper_bound` / `lower_bound`? or a multi map?

nothing critical just wondered if the code would be easier to read

Reply:
good idea, a multimap makes the code a lot more clear actually